src/DigitalSubscriberAppBanner.test.tsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { DigitalSubscriberAppBanner } from './DigitalSubscriberAppBanner';
+
+describe('DigitalSubscriberAppBanner', () => {
+    describe('when a button is clicked', () => {
+        const baseProps = () => ({
+            logButtonClickWithBraze: jest.fn(),
+            submitComponentEvent: jest.fn(),
+            brazeMessageProps: {
+                header: 'A note to our digital subscribers',
+                body: 'Hi John, did you know...',
+            },
+        });
+
+        it('invokes logButtonClickWithBraze with the internal button ID', () => {
+            const logButtonClickWithBraze = jest.fn();
+            const { getByText } = render(
+                <DigitalSubscriberAppBanner
+                    {...baseProps()}
+                    logButtonClickWithBraze={logButtonClickWithBraze}
+                />,
+            );
+
+            fireEvent.click(getByText('Ok, got it'));
+
+            expect(logButtonClickWithBraze).toHaveBeenCalledWith(0);
+        });
+
+        it('invokes submitComponentEvent with correct data', () => {
+            const logButtonClickWithOphan = jest.fn();
+            const { getByText } = render(
+                <DigitalSubscriberAppBanner
+                    {...baseProps()}
+                    submitComponentEvent={logButtonClickWithOphan}
+                />,
+            );
+
+            fireEvent.click(getByText('Ok, got it'));
+
+            expect(logButtonClickWithOphan).toHaveBeenCalledWith({
+                component: {
+                    componentType: 'RETENTION_ENGAGEMENT_BANNER',
+                    id: 'DigitalSubscriberAppBanner',
+                },
+                action: 'CLICK',
+                value: '1',
+            });
+        });
+
+        it('closes the banner', () => {
+            const props = baseProps();
+            const { getByText, queryByText } = render(<DigitalSubscriberAppBanner {...props} />);
+
+            fireEvent.click(getByText('Ok, got it'));
+
+            expect(queryByText(props.brazeMessageProps.header)).toBe(null);
```

Comment:
It's not obvious to me as someone inexperienced with React that checking for null properties is the same as checking the component has been hidden/removed. If this is obvious to Reactors then no worries at all, but if not is there a more direct way to check if the component is no long visible?

Reply:
This is a good point. I know from experience having tried to do this in the past that it isn't trivial to expect that nothing was rendered, which is basically what we're after here. Testing for the lack of header and body was an imperfect proxy for that. I'll dig a bit more, maybe I've missed something.

Reply:
No harm if this is the best way, but going on what you've said here it might be worth adding a comment that summarises what you have said above?